### DISCUSSION TITLE
how to modify state when form is submitted

### DISCUSSION THREAD

**Discussion by G3z** [filamentphp/filament#2516] [Help] [answered]
I created a simple Input to show money value (`100.00`) but save them as cents (`100000`) to avoid problems with floats in db.
This works when creating and saving but after saved the state gets updated and `10000` is shown in the input how can I intercept.

I know about `TextInput::make('cost')->mask(fn (TextInput\Mask $mask) => $mask->money('$', ',', 2))` but that is enforced in JS and does funky things when updating

Is my approach wrong ?

```php
namespace App\Forms\Components;

use Filament\Forms\Components\TextInput;

class MoneyInput extends TextInput
{
	public static function make(string $name, string $currency = '€'): static
	{
		return parent::make($name)
			->lazy()
			->prefix($currency)
			->afterStateHydrated(fn (self $component, $state) => $component->state($state / 100))
			->dehydrateStateUsing(fn ($state) => $state * 100);
	}
}
```

**Comment by danharrin** [comment, marked as answer]
Use an attribute cast on your model?

Your custom field approach isn't wrong though, I just think this should be handled by your model and not in your form
